Source/DafnyCore/Backends/Rust/Dafny-compiler-rust-definitions.dfy

```diff
@@ -24,6 +24,13 @@ module {:extern "Defs"} DafnyToRustCompilerDefinitions {
     "box","do","final","macro","override","priv","try","typeof","unsized",
     "virtual","yield"}
 
+  // Method names that would automatically resolve to trait methods instead of inherent methods
+  // Hence, full name is always required for these methods
+  const builtin_trait_preferred_methods := {
```

Comment:
OOC: why is the shape of the call different between trait and inherent methods?

Reply:
I think this is because the trait `PartialOrd` is one of the few traits that is automatically in scope _and_ it provides implementations for some functions.
Normally, Dafny methods are defined only once, either once in the trait hierarchy or in the final struct.
However, `PartialOrd` defines the `le`, `lt`, `ge`, `gt` methods based on `cmp`. Because `PartialOrd` is in scope, its methods take priority over inherent methods if using an ambiguous infix syntax. Our Dafny newtypes implement `PartialOrd`, so in the special case when someone actually defines a method like the one above, the call has to be fully specified.
More information here. There are a few subtleties I don't fully understand but I know what this change fixes bad behavior.
https://doc.rust-lang.org/reference/expressions/method-call-expr.html